In [1]:
import numpy as np
import pandas as pd
import scipy.stats as stats

Задача 1.

Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks): zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков, а затем с использованием функций из библиотек numpy и pandas.

In [2]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

In [3]:
cov_coef = np.mean(zp*ks) - np.mean(zp)*np.mean(ks)
cov_coef

9157.839999999997

In [4]:
np.cov(zp,ks, ddof=0)

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

In [5]:
r = cov_coef/(np.std(zp)*np.std(ks))
r

0.8874900920739158

In [6]:
np.corrcoef(zp,ks)

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

Задача 2.

Измерены значения IQ выборки студентов, обучающихся в местных технических вузах:

131, 125, 115, 122, 131, 115, 107, 99, 125, 111.

Известно, что в генеральной совокупности IQ распределен нормально.

Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [7]:
iq = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
alfa = 0.05
t = stats.t.ppf (0.975,9)

Так как нам неизвестно среднее квадратичное отклонение генеральной совокупности, то доверительный интервал будем строить по критерию Стьюдента.

In [8]:
x_0 = np.mean(iq) - t*np.std(iq, ddof=1)/np.sqrt(len(iq))
x_0

110.55608365158724

In [9]:
x_1 = np.mean(iq) + t*np.std(iq, ddof=1)/np.sqrt(len(iq))
x_1

125.64391634841274

Доверительный интервал [110.56; 125.64]

Задача 3.

Известно, что рост футболистов в сборной распределен нормально с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27, среднее выборочное составляет 174.2. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [10]:
sigma = 5
n = 27
avg = 174.2
alpha = 0.05

In [11]:
x_0 = avg - 1.96*sigma/np.sqrt(n)
x_0

172.31398912064722

In [12]:
x_1 = avg + 1.96*sigma/np.sqrt(n)
x_1

176.08601087935276

Ответ: Доверительный интервал [172.31; 176.09]